merlin/schema/schema.py

```diff
@@ -240,12 +264,14 @@ def value_count(self) -> Optional[Domain]:
         return Domain(**value_count) if value_count else None
 
     def __merge__(self, other):
-        col_schema = self.with_tags(other.tags)
-        col_schema = col_schema.with_properties(other.properties)
-        col_schema = col_schema.with_dtype(
-            other.dtype, is_list=other.is_list, is_ragged=other.is_ragged
+        col_schema = (
+            self.with_name(other.name)
+            .with_dtype(other.dtype)
+            .with_tags(other.tags)
+            .with_properties(other.properties)
+            .with_properties({"is_list": other.is_list, "is_ragged": other.is_ragged})
         )
-        col_schema = col_schema.with_name(other.name)
+        # TODO: Figure out if and how to merge shapes here too
```

Comment:
Is this deliberately left to future work or do we need a way to merge shapes still?

Reply:
Yeah, this should get filled in. The tricky part is figuring out what it means to merge shapes. 😅

Since you can have unbounded dimensions with `min=0, max=None`, one possible way to define it would be to take the shape from each column schema and:
- Check that the shapes have the same number of dimensions, if not raise an error
- For each dimension:
  - Check if the two versions of the dim are identical, if so use either one
  - Check if one is bounded and the other is unbounded, if so use the bounded version
  - Otherwise, we have two bounded versions of the dim that don't match, and we can either:
    - Choose whether the left or right side of the merge gets priority and overwrites the other
    - Raise an error

Reply:
For now, I'm just going to give the same priority to the shape from `other` as the rest of the attributes. We'll see if that causes problems and adjust if necessary.